src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,281 @@
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(self.client, self.id).parsed
+        return self.__users
+
+    def __str__(self) -> str:
+        return (
+            f"Workspace(id={self.id}, name={self.name}, users={self.users}, insterted_at={self.inserted_at},"
+            f" updated_at={self.updated_at}"
+        )
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
+    def add_user(self, id: str) -> None:
+        """Adds an existing user to the workspace in Argilla.
+
+        Args:
+            id: the ID of the user to be added to the workspace. The user must exist in Argilla.
+
+        Raises:
+            ValueError: if the user with the provided ID already exists in the workspace.
+            RuntimeError: if there was an error while adding the user to the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.from_name("my-workspace")
+            >>> workspace.add_user("my-user-id")
+        """
+        try:
+            created_user = workspaces_api.create_workspace_user(
+                client=self.client,
+                id=self.id,
+                user_id=id,
+            )
+        except AlreadyExistsApiError as e:
+            raise ValueError(f"User with id=`{id}` already exists in workspace with id=`{self.id}`.") from e
+        except BaseClientError as e:
+            raise RuntimeError(f"Error while adding user with id=`{id}` to workspace with id=`{self.id}`.") from e
+
+        if len(self.users) == 0:
+            self.__users = [created_user]
+        else:
+            self.__users.append(created_user)
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
+    def delete_user(self, id: str) -> None:
+        """Deletes an existing user from the workspace in Argilla. Note that the user
+        will not be deleted from Argilla, but just from the workspace.
+
+        Args:
+            id: the ID of the user to be deleted from the workspace. The user must exist in Argilla.
+
+        Raises:
+            ValueError: if the user with the provided ID doesn't exist in the workspace.
+            RuntimeError: if there was an error while deleting the user from the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.from_name("my-workspace")
+            >>> workspace.delete_user("my-user-id")
+        """
+        try:
+            workspaces_api.delete_workspace_user(
```

Comment:
same here, maybe we need to find it in list before, just for early validations

Reply:
So same response as above, is it worth to have another API call to repeat a validation that is already included as part of the endpoints we're calling? Isn't just better to catch the exceptions and show the relevant error messages based on those?